File: js/day2.js
import fs from "fs";

const reader = (path) => {
  const text = fs.readFileSync(path, "utf-8");
  const lines = text.trim().split(/\r?\n/);
  return lines;
};

const dayTwo = (input) => {
  const arr = [];
  let counter = 0;
  for (let i = 0; i < input.length; i++) {
    const array = input[i].split("x");
    arr.push(array);
  }
  for (let i = 0; i < arr.length; i++) {
    counter = calculator(arr[i]) + counter;
  }
  return counter;
};

function calculator(params) {
  const min = Math.min(
    params[0] * params[1],
    params[1] * params[2],
    params[0] * params[2],
  );
  const nums =
    2 * params[0] * params[1] +
    2 * params[1] * params[2] +
    2 * params[0] * params[2];
  return min + nums;
}

console.log(dayTwo(reader("../day2.txt")));

File: python/day2.py
def reader(foldername): 
    with open(foldername, encoding="utf-8") as f:
        data = f.read().splitlines()
    return data


def calc(params):
    a, b, c = map(int, params)
    min_num = min(a*b, b*c, a*c)
    nums = 2*(a*b + b*c + a*c)
    return min_num + nums


def calc_day_two(params): 
    a, b, c = sorted(map(int, params))
    wrap = 2 * (a + b)
    bow = a * b * c
    return wrap + bow


def day_two(input_list): 
    arr = []
    counter = 0
    for item in input_list: 
        array = item.split("x")      
        arr.append(array)
    for dims in arr:                 
        counter += calc_day_two(dims)        
    return counter


input_day_two = reader("../day2.txt")



print(day_two(input_day_two))



